services/QuillLMS/app/queries/prompt_feedback_history.rb

```diff
@@ -18,36 +25,126 @@ def self.promptwise_sessions(activity_id)
 
     def self.promptwise_postprocessing(grouped_feedback_histories)
         prompt_hash = {}
-
+        
         prompts = Comprehension::Prompt.where(id: grouped_feedback_histories.map(&:prompt_id))
    
         grouped_feedback_histories.each do |prompt_session|
-            prompt_id = prompt_session.prompt_id
+            prompt_id = prompt_session.prompt_id.to_s.to_sym
 
             prompt_hash[prompt_id] ||= {
-                optimal_final_attempts: 0, 
-                session_count: 0, 
-                total_responses: 0,
-                final_attempt_pct_optimal: 0,
-                final_attempt_pct_not_optimal: 0,
-                display_name: ''
+
+                total_responses: 0.0,
+                session_count: 0.0, 
+                optimal_attempt_array: [],
+                display_name: '',
+
+                pct_final_attempt_optimal: 0.0,
+                pct_final_attempt_not_optimal: 0.0,
+                num_final_attempt_optimal: 0.0,
+                num_final_attempt_not_optimal: 0.0,
+
+                avg_attempts_to_optimal: 0.0,
+
+                num_consecutive_repeated_attempts_for_same_rule: 0.0,
+                num_non_consecutive_repeated_attempts_for_same_rule: 0.0,
+                pct_consecutive_repeated_attempts_for_same_rule: 0.0,
+                pct_non_consecutive_repeated_attempts_for_same_rule: 0.0,
+
+                num_first_attempt_optimal: 0.0,
+                num_first_attempt_not_optimal: 0.0,
+                pct_first_attempt_optimal: 0.0,
+                pct_first_attempt_not_optimal: 0.0
```

Comment:
Agreed!

Reply:
👍

Reply:
I've added the API payload expectation here: https://www.notion.so/quill/Backend-New-Prompt-Health-Report-Table-MVP-c743eb47e7e547b7868d5e8ba2e350af#c94135e5d4704996bc3056d5e7fa8ccb

Reply:
I think what @anathomical is suggesting, and what I would prefer, is that that documentation live in the documentation folder (like his example) rather than in the task card, which can be harder to find later for future reference and is also weirder to change (if you have a new task that modifies the API, for instance).